pkg/util/list.go

```diff
@@ -0,0 +1,106 @@
+package util
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strconv"
+)
+
+// ListIterator lists the objects from the given url by loading one page at a time
+type ListIterator struct {
```

Comment:
User provided query parameters currently seems to not being supported

Reply:
query parameters can be passed in the url
```go
u, err := url.Parse(baseurl)
q := u.Query()
q.Add("key", "value")
u.RawQuery = q.Encode()
fullurl := u.String()
```